conda_lock/conda_lock.py

```diff
@@ -552,106 +401,91 @@ def make_lock_files(
         )
 
 
-def is_micromamba(conda: PathLike) -> bool:
-    return str(conda).endswith("micromamba") or str(conda).lower().endswith(
-        "micromamba.exe"
-    )
-
-
-def create_lockfile_from_spec(
+def render_lockfile_for_platform(  # noqa: C901
     *,
-    conda: PathLike,
-    spec: LockSpecification,
+    lockfile: Lockfile,
+    include_dev_dependencies: bool,
+    extras: Optional[AbstractSet[str]],
     kind: str,
+    platform: str,
 ) -> List[str]:
-    assert spec.virtual_package_repo is not None
-    virtual_package_channel = spec.virtual_package_repo.channel_url
-    dry_run_install = solve_specs_for_arch(
-        conda=conda,
-        platform=spec.platform,
-        channels=[*spec.channels, virtual_package_channel],
-        specs=spec.specs,
-    )
-    logging.debug("dry_run_install:\nspec: %s\nresult: %s", spec, dry_run_install)
 
     lockfile_contents = [
         "# Generated by conda-lock.",
-        f"# platform: {spec.platform}",
-        f"# input_hash: {spec.input_hash()}\n",
+        f"# platform: {platform}",
+        f"# input_hash: {lockfile.metadata.content_hash}\n",
     ]
 
+    categories = {
+        "main",
+        *(extras or []),
+        *(["dev"] if include_dev_dependencies else []),
+    }
+
+    conda_deps = []
+    pip_deps = []
+    for p in lockfile.package:
+        if p.platform == platform and ((not p.optional) or (p.category in categories)):
+            if p.manager == "pip":
+                pip_deps.append(p)
+            # exclude virtual packages
+            elif not p.name.startswith("__"):
+                conda_deps.append(p)
+
+    def format_pip_requirement(
+        spec: LockedDependency, platform: str, direct=False
+    ) -> str:
+        if spec.source and spec.source.type == "url":
+            return f"{spec.name} @ {spec.source.url}"
+        elif direct:
+            return f'{spec.name} @ {spec.url}#{spec.hash.replace(":", "=")}'
+        else:
+            return f"{spec.name} === {spec.version}"
```

Comment:
That's a good point; I wasn't sure conda actually implements the pip install stage. Since it does it from a synthesized requirements file, it looks like the hashes can be added there. Done.